src/adapters/hasura/altStudent.adapter.ts

```diff
@@ -1041,4 +1050,138 @@ export class ALTStudentService {
       data: responseData,
     });
   }
+  public async updateStudent(userId, request, body) {
+    const decoded: any = jwt_decode(request.headers.authorization);
+    const altUserRoles =
+      decoded["https://hasura.io/jwt/claims"]["x-hasura-allowed-roles"];
+
+    const studentFields = [
+      "groups",
+      "religion",
+      "caste",
+      "annualIncome",
+      "motherEducation",
+      "fatherEducation",
+      "motherOccupation",
+      "fatherOccupation",
+      "noOfSiblings",
+      "schoolUdise",
+      "board",
+      "state",
+      "block",
+      "district",
+    ];
+
+    const userFields = ["name", "email", "gender", "dateOfBirth", "mobile"];
+    let userUpdate = "";
+    let studentUpdate = "";
+    let userUpdateFields = "";
+    let studentUpdateFields = "";
+
+    Object.keys(body).forEach((field) => {
+      if (body[field] !== "" && studentFields.includes(field)) {
+        studentUpdate += `${field}: ${
+          typeof body[field] === "string"
+            ? `"${body[field]}"`
+            : JSON.stringify(body[field])
+        }, `;
+        studentUpdateFields += `${field} `;
+      }
+    });
+
+    // Construct the userUpdate string
+    Object.keys(body).forEach((field) => {
+      if (body[field] !== "" && userFields.includes(field)) {
+        userUpdate += `${field}: ${
+          typeof body[field] === "string"
+            ? `"${body[field]}"`
+            : JSON.stringify(body[field])
+        }, `;
+        userUpdateFields += `${field} `;
+      }
+    });
+
+    const data = {
+      query: `mutation MyMutation($userId: uuid) {
+        ${
+          studentUpdate
```

Comment:
@YoginiTayade Will the update query for student table be executed even if there was not data provided fro update of student table? Ideally it should not be executed in this case.

If the record is updated then respective records updatedBy and updatedAt columns should also be updated.

Reply:
resolved